src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -93,9 +98,27 @@ private Func<
         private volatile int _disposed;
         private SafeWinHttpHandle? _sessionHandle;
         private readonly WinHttpAuthHelper _authHelper = new WinHttpAuthHelper();
+        private readonly Timer? _certificateCleanupTimer;
+        private bool _isTimerRunning;
+        private ConcurrentDictionary<CachedCertificateKey, CachedCertificateValue> _cachedCertificates = new();
```

Comment:
```suggestion
        private readonly ConcurrentDictionary<CachedCertificateKey, CachedCertificateValue> _cachedCertificates = new();
```

Nit: consider allocating this in the ctor behind the flag to make it more pay-for-play